libtalek/client_test.go

```diff
@@ -126,3 +126,42 @@ func TestRead(t *testing.T) {
 		t.Fatalf("Read wasn't for the enqueued subscription. %v / %v / %d", rv1, rv2, bucket)
 	}
 }
+
+func TestFragment(t *testing.T) {
+	theMsg := make([]byte, 1024*256)
+	validateMsg := make([]byte, 1024*256)
+
+	hdr := msgHeader{}
+	hdr.left = uint32(len(theMsg))
+
+	parts := make([][]byte, 0)
+	for hdr.left > 0 {
```

Comment:
Also, in theory, it is possible to get partials out of order, if you have a retransmission or something (yea I know thats not implemented yet). So I think you'd need enough state to be able to recompose messages under any ordering and provide enough state to know what to request for a retransmission.